operate/services/manage.py

```diff
@@ -232,6 +237,44 @@ def create(
 
         return service
 
+    def _compute_service_env_variables(
+        self, service: Service, staking_params: t.Dict[str, str]
+    ) -> None:
+        """Compute values to override service.yaml variables for the deployment."""
+
+        # TODO A customized, arbitrary computation mechanism should be devised.
+        computed_values = {
+            "ETHEREUM_LEDGER_RPC": PUBLIC_RPCS[ChainType.ETHEREUM],
+            "GNOSIS_LEDGER_RPC": PUBLIC_RPCS[ChainType.GNOSIS],
+            "BASE_LEDGER_RPC": PUBLIC_RPCS[ChainType.BASE],
+            "OPTIMISM_LEDGER_RPC": PUBLIC_RPCS[ChainType.OPTIMISM],
+            "STAKING_CONTRACT_ADDRESS": staking_params.get("staking_contract"),
+            "MECH_ACTIVITY_CHECKER_CONTRACT": staking_params.get("activity_checker"),
+            "MECH_CONTRACT_ADDRESS": staking_params.get("agent_mech"),
+            "MECH_REQUEST_PRICE": "10000000000000000",
+            "USE_MECH_MARKETPLACE": "mech_marketplace"
+            in service.chain_configs[
+                service.home_chain_id
+            ].chain_data.user_params.staking_program_id,
```

Comment:
@0xArdi Is this approach correct? I mean, if the service is staked in multiple chains, and we could have different combinations of mech marketplace or not, what value should this environment variable have? Currently it is set to `true` if the home chain staking contract is a mech_marketplace one.

Reply:
I think the `mech_interact_abci` only allows for the agent to work with the mech in a single chain. 
This param [here](https://github.com/valory-xyz/IEKit/blob/main/packages/valory/skills/mech_interact_abci/skill.yaml#L151) might tell us what chain the mech is meant to be used on. Perhaps that needs to be checked instead of assuming its always the home_chain_id.